modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -149,16 +149,18 @@ public void setFlowConstruct(FlowConstruct flowConstruct)
 
     private MetadataContext getMetadataContext() throws MetadataResolvingException
     {
-        //TODO MULE-9530: Improve Config retrieval for Metadata resolution
-        if (!StringUtils.isBlank(configurationProviderName) &&
-            muleContext.getRegistry().get(configurationProviderName) instanceof DynamicConfigurationProvider)
+        MuleEvent fakeEvent = getInitialiserEvent(muleContext);
+        ConfigurationInstance<Object> configuration = getConfiguration(fakeEvent);
```

Comment:
yeah, probably the `.orElseThrow(()` will never be called and could be replaced by a plain `get`